content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol/index.md

```diff
@@ -0,0 +1,144 @@
+- [Preparation](#preparation)
+- [Create a Connection](#create-a-connection)
+- [Create a Command](#create-a-command)
+- [Read the Data](#read-the-data)
+- [Housekeeping](#housekeeping)
+- [Other Data Providers](#other-data-providers)
+- [Wrap-Up](#wrap-up)
+
+### Basic ADO Flow
+What do I do now that Microsoft has provided me with this one-of-a-kind library of data access procedures? How can I get started? What do I look for when I am looking for a place to stay? We will start by looking at `SQL Server's namespace`, which governs data access. At the end of this article, we will go over different data providers, but for now, we will stick with SQL Server. System.Data.SqlClient is the namespace to remember if you want to connect to SQL Server.
+
+> Keep this namespace in mind. It's critical!
+
+There are presently 16 classes in the SQLClient namespace. These classes provide a programmatic way to establish connections to databases, verify permissions, manage exceptions, and do other duties related to data access. Let's get started using ADO.
+
+To view data object, follow this basic procedure:
+1. Establish a link to the data object.
+2. Make a command that the data object will follow.
+3. Consider the data that the data object has provided.
+
+To complete the actions listed above, we will need to employ the following classes:
+1. System.Data.SqlClient.SqlConnection
+2. System.Data.SqlClient.SqlCommand
+3. System.Data.SqlClient.SqlDataReader
+
+### Preparation
+For this project, I used a safe console application. To begin, we must declare the variables that will be used. We will need to make a variable to hold the data, one to hold the connection attributes, and another to hold the SQL statements we will be issuing. Any exceptions that we encounter when reading from the database are noted. In the working-storage section, all of these items are listed below.
+
+```SQL
+working-storage section.
+sqlConnection type System.Data.SqlClient.SqlConnection.
+sqlDataReader type System.Data.SqlClient.SqlDataReader.
+sqlCommand    type System.Data.SqlClient.SqlCommand.
+```
+These are the essential objects that will allow us to connect to the database engine and use ADO, as well as get information from the database engine about the operations we run there.
+
+A few functional fields will be added next, as well as a few fields for storing data we'll be retrieving from the database. The following are the contents of the completed working-storage section:
+
+
+```sql
+working-storage section.
+sqlConnection type System.Data.SqlClient.SqlConnection.
+sqlDataReader type System.Data.SqlClient.SqlDataReader.
+sqlCommand    type System.Data.SqlClient.SqlCommand.
+
+sqlCmd             string.
+dataFound          type System.Boolean.
+recordsRead        pic 9(06)  value zeros.
+LastName           pic x(20)  value spaces.
+FirstName          string.
+HomePhone          pic x(24)  value spaces.    
+junk               pic x(01)  value spaces.

```

Comment:
Is this some kind of output? If so, add this under a separate code snippet

Reply:
This is not an output, we are declaring the variables in that code snippet.